packages/sdk-browser/src/sdk-client.ts

```diff
@@ -217,15 +234,63 @@ export class TurnkeyBrowserSDK {
   logoutUser = async (): Promise<boolean> => {
     await removeStorageValue(StorageKeys.AuthBundle); // DEPRECATED
     await removeStorageValue(StorageKeys.CurrentUser);
+    await removeStorageValue(StorageKeys.UserSession);
     await removeStorageValue(StorageKeys.ReadWriteSession);
 
     return true;
   };
 }
 
 export class TurnkeyBrowserClient extends TurnkeySDKClientBase {
-  constructor(config: TurnkeySDKClientConfig) {
+  authClient: AuthClient | undefined;
+
+  constructor(config: TurnkeySDKClientConfig, authClient?: AuthClient) {
     super(config);
+    this.authClient = authClient;
+  }
+
+  async saveSession({
+    organizationId,
+    organizationName,
+    sessionExpiry,
+    credentialBundle,
+    userId,
+    username,
+    ...sessionResponse
+  }: SessionResponse) {
+    if (!this.authClient) {
+      throw new Error("Failed to save session: Authentication client not set");
+    }
+
+    const expiry = Number(sessionExpiry);
+    const session = credentialBundle
+      ? {
+          write: {
+            credentialBundle,
+            expiry,
+          },
+        }
+      : {
+          read: {
+            token: sessionResponse.session!,
+            expiry,
+          },
+        };
+
+    const userSession: User = {
+      userId,
+      username,
+      organization: {
+        organizationId,
+        organizationName,
+      },
+      session: {
+        authenticatedClient: this.authClient,
```

Comment:
also open to calling `authClient --> authenticatedClient`. don't feel tooo strongly on this though